addOns/oast/src/main/java/org/zaproxy/addon/oast/OastInsertPayloadMenu.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2023 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.oast;
+
+import java.awt.Component;
+import java.util.Map;
+import javax.swing.JMenu;
+import javax.swing.JMenuItem;
+import javax.swing.text.JTextComponent;
+import org.parosproxy.paros.Constant;
+import org.zaproxy.zap.extension.ExtensionPopupMenu;
+
+@SuppressWarnings("serial")
+public class OastInsertPayloadMenu extends ExtensionPopupMenu {
+    private static final long serialVersionUID = 2L;
+    private JTextComponent lastInvoker;
+    private ExtensionOast extension;
+
+    public OastInsertPayloadMenu(ExtensionOast extension) {
+        super();
+        this.extension = extension;
+        setText(Constant.messages.getString("oast.popup.menu.insertPayload"));
+        add(new OastInsertPayloadHandler(this, true));
+        add(new OastInsertPayloadHandler(this, false));
+    }
+
+    private class OastInsertPayloadHandler extends JMenu {
+        private OastInsertPayloadMenu parentMenu;
```

Comment:
Not necessary, `lastInvoker` can now be accessed directly.

Reply:
This was only partially addressed, `parentMenu` is still being used while the `lastInvoker` can be accessed directly.